tests/ga/test_agent_update_handler.py

```diff
@@ -446,6 +449,29 @@ def test_it_should_report_update_status_with_success(self):
             self.assertEqual(0, vm_agent_update_status.code)
             self.assertEqual(str(CURRENT_VERSION), vm_agent_update_status.expected_version)
 
+    def test_it_should_not_report_update_status_when_self_update_used(self):
+        self.prepare_agents(count=1)
+
+        data_file = DATA_FILE.copy()
+        data_file['ext_conf'] = "wire/ext_conf.xml"
+        with self._get_agent_update_handler(test_data=data_file) as (agent_update_handler, _):
+            with self.assertRaises(AgentUpgradeExitException):
+                agent_update_handler.run(agent_update_handler._protocol.get_goal_state(), True)
+        vm_agent_update_status = agent_update_handler.get_vmagent_update_status()
+        self.assertIsNone(vm_agent_update_status, "VM Agent Update Status should not be set when self-update is used")
+
+    def test_it_should_report_update_with_error_if_auto_update_is_disabled(self):
```

Comment:
it should only report update with error if auto update is disabled and the update mode is rsm, right?
Right now the test name sounds like it should always be reported if auto update is disabled

Reply:
updated